src/nvme/private.h

```diff
@@ -160,15 +160,19 @@ struct nvme_fabric_options {
 	bool trsvcid;
 };
 
+struct nvme_log {
+	FILE *fp;
+	int level;
+	bool pid;
+	bool timestamp;
+};
+
 struct nvme_root {
 	char *config_file;
 	char *application;
 	struct list_head hosts;
 	struct list_head endpoints; /* MI endpoints */
-	FILE *fp;
-	int log_level;
-	bool log_pid;
-	bool log_timestamp;
+	struct nvme_log log;
```

Comment:
Is this still needed since seems only the default log data added enough and duplicated. Also if needed seems nvme_root can have a pointer to the default log data.

Reply:
Depends. If we want to support different log levels per root object then we should keep it if not then we could rip this part out as well and reduce the complexity. 

Not sure though as it is changing the behavior a bit.

Reply:
I see. To make sure let me confirm again. Multiple root objects supported or in future possible to be supported? (I would like confirm the possibility since if no possibility only the default log data enough and the behavior for use not changed I think.)

Reply:
We do create multiple root objects in libblockdev since it's often used in threads and need separate memory objects. While not currently used, it might be useful to grab the log for certain operations and present it as a buffer back to the user (e.g. via D-Bus). Besides, as backwards compatibility is concerned, I would keep it there. Reconsider later for libnvme 2.0?

Reply:
Understood. Thanks for your explanation.

Reply:
@tbzatek thanks for the feedback. Good to know that you are using this feature. In this case, we keep it as it is, just have a global default fallback for all commands. Not nice, but I suppose the ship has sailed and we have to live with it.

Anyway, when we do libnvme 2.0 we could introduce a generic handle for all APIs. This should solve the issue for good.  The downside is that you always have to open/create a handle, though I think this is an acceptable requirement.